src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/CompletionSet2.cs

```diff
@@ -101,13 +108,14 @@ internal void SetTrackingSpan(ITrackingSpan trackingSpan)
         private VSCompletion ConvertCompletionItem(CompletionItem item, string displayTextOpt = null)
         {
             VSCompletion value;
-            if (!_completionItemMap.TryGetValue(item, out value))
+            if (!_completionItemToVSCompletion.TryGetValue(item, out value))
             {
                 value = new CustomCommitCompletion(
                     _completionPresenterSession,
                     item,
                     displayTextOpt ?? item.DisplayText);
-                _completionItemMap.Add(item, value);
+                _completionItemToVSCompletion.Add(item, value);
+                value.Properties.AddProperty(typeof(CompletionItem), item);
```

Comment:
Yes.  I'm saying i'll switch away from teh property bag to a reverse dictionary of our own.  I'm of the beleif that if we have one dictionary from K to V, then adding another dictionary from V to K is fine.